Failing bankKeeper dispatch in Wordle tutorial

### Version of Git SHA

0ce8e57ed2c34ef4d30b6e24a13ac3f9feaddf31

### System OS

Linux scw-boring-curie 5.15.0-94-generic #104-Ubuntu SMP Tue Jan 9 15:25:40 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

### Installed tools

_No response_

### Others

_No response_

### Steps to reproduce it

Follow tutorial, run any `wordled` command from the `init.sh` script

### Expected result

Should end up running rollup

### Actual result

```
panic: can't resolve type wordle/x/wordle/types/types.BankKeeper for wordle/x/wordle/module.ProvideModule (/root/wordle/x/wordle/module/module.go:194):
	while resolving:
		types.BankKeeper for wordle/x/wordle/module.ProvideModule (/root/wordle/x/wordle/module/module.go:194)
		store.KVStoreService for wordle/x/wordle/module.ProvideModule (/root/wordle/x/wordle/module/module.go:194)
		store.KVStoreService for github.com/cosmos/cosmos-sdk/x/consensus.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/consensus/module.go:117)
		exported.Subspace for github.com/cosmos/cosmos-sdk/x/crisis.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/crisis/module.go:201)
		types.AppOptions for github.com/cosmos/cosmos-sdk/x/crisis.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/crisis/module.go:201)
		store.KVStoreService for github.com/cosmos/cosmos-sdk/x/crisis.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/crisis/module.go:201)
		store.KVStoreService for github.com/cosmos/cosmos-sdk/x/authz/module.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/authz/module/module.go:178)
		store.KVStoreService for cosmossdk.io/x/circuit.ProvideModule (/root/go/pkg/mod/cosmossdk.io/x/circuit@v0.1.0/module.go:154)
		exported.Subspace for github.com/cosmos/cosmos-sdk/x/mint.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/mint/module.go:221)
		types.InflationCalculationFn for github.com/cosmos/cosmos-sdk/x/mint.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/mint/module.go:221)
		store.KVStoreService for github.com/cosmos/cosmos-sdk/x/mint.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/mint/module.go:221)
		depinject.OwnModuleKey for github.com/cosmos/cosmos-sdk/x/mint.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/mint/module.go:221)
		exported.Subspace for github.com/cosmos/cosmos-sdk/x/slashing.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/slashing/module.go:230)
		store.KVStoreService for github.com/cosmos/cosmos-sdk/x/slashing.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/slashing/module.go:230)
		types.ParamSubspace for github.com/cosmos/cosmos-sdk/x/gov.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/gov/module.go:189)
		store.KVStoreService for github.com/cosmos/cosmos-sdk/x/distribution.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/distribution/module.go:232)
		types.DistributionKeeper for github.com/cosmos/cosmos-sdk/x/gov.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/gov/module.go:189)
		store.KVStoreService for github.com/cosmos/cosmos-sdk/x/staking.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/staking/module.go:221)
		types.StakingKeeper for github.com/cosmos/cosmos-sdk/x/gov.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/gov/module.go:189)
		depinject.OwnModuleKey for github.com/cosmos/cosmos-sdk/x/gov.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/gov/module.go:189)
		store.KVStoreService for github.com/cosmos/cosmos-sdk/x/gov.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/gov/module.go:189)
		types.AppOptions for cosmossdk.io/x/upgrade.ProvideModule (/root/go/pkg/mod/cosmossdk.io/x/upgrade@v0.1.1/module.go:194)
		store.KVStoreService for cosmossdk.io/x/upgrade.ProvideModule (/root/go/pkg/mod/cosmossdk.io/x/upgrade@v0.1.1/module.go:194)
		map[string]appmodule.AppModule for wordle/cmd/wordled/cmd.NewRootCmd (/root/wordle/cmd/wordled/cmd/root.go:41)
		ante.FeegrantKeeper for github.com/cosmos/cosmos-sdk/x/auth/tx/config.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/auth/tx/config/config.go:70)
		depinject.ModuleKey for github.com/cosmos/cosmos-sdk/x/params.ProvideSubspace (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/params/module.go:146)
		exported.Subspace for github.com/cosmos/cosmos-sdk/x/auth.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/auth/module.go:209)
		func() types.AccountI for github.com/cosmos/cosmos-sdk/x/auth.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/auth/module.go:209)
		types.RandomGenesisAccountsFn for github.com/cosmos/cosmos-sdk/x/auth.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/auth/module.go:209)
		store.KVStoreService for github.com/cosmos/cosmos-sdk/x/auth.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/auth/module.go:209)
		types.AccountKeeper for github.com/cosmos/cosmos-sdk/x/bank.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/bank/module.go:227)
		store.KVStoreService for github.com/cosmos/cosmos-sdk/x/bank.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/bank/module.go:227)
		types.BankKeeper for github.com/cosmos/cosmos-sdk/x/auth/tx/config.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/auth/tx/config/config.go:70)
		func() runtime.ValidatorAddressCodec for github.com/cosmos/cosmos-sdk/runtime.ProvideAddressCodec (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/runtime/module.go:281)
		func() address.Codec for github.com/cosmos/cosmos-sdk/runtime.ProvideAddressCodec (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/runtime/module.go:281)
		address.Codec for github.com/cosmos/cosmos-sdk/x/auth/tx/config.ProvideModule (/root/go/pkg/mod/github.com/rollkit/cosmos-sdk@v0.50.5-rollkit-v0.13.0-no-fraud-proofs/x/auth/tx/config/config.go:70)
		tx.ConfigOptions for wordle/cmd/wordled/cmd.NewRootCmd (/root/wordle/cmd/wordled/cmd/root.go:41)


goroutine 1 [running]:
wordle/cmd/wordled/cmd.NewRootCmd()
	/root/wordle/cmd/wordled/cmd/root.go:56 +0x7d7
main.main()
	/root/wordle/cmd/wordled/main.go:14 +0x17
```
	

### Relevant log output

_No response_

### Notes

_No response_

--- a/tutorials/wordle.md
+++ b/tutorials/wordle.md
@@ -358,8 +358,7 @@ func (k msgServer) SubmitWordle(goCtx context.Context, msg *types.MsgSubmitWordl
   reward := sdk.Coins{sdk.NewInt64Coin("token", 100)}
   // Escrow Reward
   submitterAddress, _ := sdk.AccAddressFromBech32(msg.Creator)
-  moduleAcct := sdk.AccAddress(crypto.AddressHash([]byte(types.ModuleName)))
-  err := k.bankKeeper.SendCoins(ctx, submitterAddress, moduleAcct, reward)
+  err := k.bankKeeper.SendCoinsFromAccountToModule(ctx, submitterAddress, types.ModuleName, reward)
   if err != nil {
     return nil, err
   }
@@ -483,9 +482,8 @@ func (k msgServer) SubmitGuess(goCtx context.Context, msg *types.MsgSubmitGuess)
     reward := sdk.Coins{sdk.NewInt64Coin("token", 100)}
     // If Submitter Guesses Correctly
     guesserAddress, _ := sdk.AccAddressFromBech32(msg.Creator)
-    moduleAcct := sdk.AccAddress(crypto.AddressHash([]byte(types.ModuleName)))
     // Send Reward
-    err := k.bankKeeper.SendCoins(ctx, moduleAcct, guesserAddress, reward)
+    err := k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, guesserAddress, reward)
     if err !=nil {
       return nil, err
     }
@@ -535,7 +533,8 @@ interface in order to allow sending the reward to the right guesser.
 
 ```go title="x/wordle/types/expected_keepers.go"
 type BankKeeper interface {
-  SendCoins(ctx sdk.Context, fromAddr sdk.AccAddress, toAddr sdk.AccAddress, amt sdk.Coins) error
+  SendCoinsFromModuleToAccount(ctx context.Context, senderModule string, recipientAddr sdk.AccAddress, amt sdk.Coins) error
+  SendCoinsFromAccountToModule(ctx context.Context, senderAddr sdk.AccAddress, recipientModule string, amt sdk.Coins) error
 }
 ```
 

